roles/create_mce_crs/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+- name: Create Template for Namespace for imported cluster
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/01_create-ns-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for Pull Secret
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/02_create-pull-secet-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for InfraEnv
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/03_create-infraenv-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for Cluster Image Set
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/04_create-cluster-image-set-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for Agent Cluster Install
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/05_create-agent-cluster-install-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for Import Cluster (ClusterDeployment)
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/07_create-cluster-deployment-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Template for Managed Cluster
+  command: "oc apply -f {{ repo_root_path }}/roles/create_mce_crs/files/managed-cluster-template.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Create Namespace for imported cluster
+  shell: "oc process create-ns-template -p NAMESPACE={{ namespace }} | oc create -f -"
+
+- name: Create Pull Secret
+  shell: "oc process create-pullsecret-template -p NAMESPACE={{ namespace }} -p DOCKERCONFIG={{ dockerconfig }} | oc create -f -"
+
+- name: Create InfraEnv
+  shell: "oc process create-infraenv-template -p NAMESPACE={{ namespace }} -p SSHKEY='{{ sshkey }}' | oc create -f -"
+
+- name: Create Cluster Image Set
+  shell: "oc process create-cluster-image-set-template -p OCPVERSION={{ openshift_full_version }} | oc create -f -"
+
+- name: Create Agent Cluster Install
+  shell: "oc process create-agentclusterinstall-template -p NAMESPACE={{ namespace }} -p APIVIP='{{ api_vip }}' -p OCPVERSION='{{ openshift_full_version }}' -p INGRESSVIP='{{ ingress_vip }}' -p PLATFORMTYPE={{ platform_type }} -p CLUSTERNETWORK={{ cluster_network_cidr }} -p SERVICENETWORK={{ service_network_cidr }} -p SSHKEY='{{ sshkey }}'| oc create -f -"
+
+- name: Create Kubeconfig
+  command: "oc -n {{ namespace }} create secret generic {{ namespace }}-admin-kubeconfig --from-file=kubeconfig={{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Get credentials

```

Comment:
These creds are already available in a vault file. Although I suppose it doesn't hurt to retrieve them again, if the this is part of the same flow. Depending how long after the install this is run those creds might have been garbage collected.

Reply:
Lets take a multi step approach to this:

1. Check for a variable `day2_kubeamin_password` or something to that effect
2. Check for the file created in post_install
3. Attempt to retrieve the value from the assisted installer API
4. Fail and request the user set the variable.